src/MiniValidation/TypeDetailsCache.cs

```diff
@@ -218,6 +224,23 @@ private static (ValidationAttribute[]?, DisplayAttribute?, SkipRecursionAttribut
         return new(validationAttributes?.ToArray(), displayAttribute, skipRecursionAttribute);
     }
 
+    private static bool TryGetAttributesViaTypeDescriptor(PropertyInfo property, out IEnumerable<Attribute>? typeDescriptorAttributes)
```

Comment:
```suggestion
    private static bool TryGetAttributesViaTypeDescriptor(PropertyInfo property, [NotNullWhen(true)]out IEnumerable<Attribute>? typeDescriptorAttributes)
```

This should allow you to remove the `!` operator above on line 204, at least I think, from memory :smile:

Reply:
I tried this before but nullable attributes are not available in .NET Standard 2.0 - at least out-of-the-box.
I just tried with PolySharp and with that it works. It adds no NuGet or assembly dependency to it, therefore I propose to go this route.